internal/chain/reward.go

```diff
@@ -220,9 +220,24 @@ func AccumulateRewardsAndCountSigs(
 			if blockNum%RewardFrequency != RewardFrequency-1 {
 				return network.EmptyPayout, nil
 			}
-			return distributeRewardAfterAggregateEpoch(bc, state, header, beaconChain, defaultReward)
+			return distributeRewardAfterAggregateEpoch(bc,
+				state,
+				header,
+				beaconChain,
+				defaultReward,
+				// pruning only occurs at the end of the epoch
+				bc.Config().IsNoNilDelegations(header.Epoch()) && header.IsLastBlockInEpoch() && header.ShardID() == shard.BeaconChainShardID,
+			)
 		} else {
-			return distributeRewardBeforeAggregateEpoch(bc, state, header, beaconChain, defaultReward, sigsReady)
+			return distributeRewardBeforeAggregateEpoch(bc,
+				state,
+				header,
+				beaconChain,
+				defaultReward,
+				sigsReady,
+				// pruning only occurs at the end of the epoch
+				bc.Config().IsNoNilDelegations(header.Epoch()) && header.IsLastBlockInEpoch() && header.ShardID() == shard.BeaconChainShardID,
```

Comment:
The reward happens every 64 blocks, so any reward logic that happens after the pruning will need to have the new logic running right?

Reply:
Agreed and updated.